test/unity/toolkits/drawing_classifier/data_utils.hpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+#ifndef TURI_DRAWING_CLASSIFIER_TESTING_DATA_UTILS_H_
+#define TURI_DRAWING_CLASSIFIER_TESTING_DATA_UTILS_H_
+
+#include <algorithm>
+
+#include <model_server/lib/image_util.hpp>
+#include <toolkits/drawing_classifier/dc_data_iterator.hpp>
+
+namespace turi {
+namespace drawing_classifier {
+
+constexpr size_t IMAGE_WIDTH = 28;
+constexpr size_t IMAGE_HEIGHT = 28;
+
+class drawing_data_generator {
+ 
+ public:
+  /** Builds an SFrame with columns "test_image" and "test_targets".
+   *  Creates an SFrame with num_rows number of rows where each row has 
+   *  a drawing (a grayscale 28x28 image)
+   *  and a corresponding target, which is the row index modulo 
+   *  unique_labels.size().
+   */
+  drawing_data_generator(size_t num_rows,
+      const std::vector<std::string> &unique_labels)
+      : num_rows_(num_rows),
+        unique_labels_(unique_labels) {
+
+    flex_list images(num_rows_);
+    flex_list labels(num_rows_);
+    
+    std::vector<unsigned char> buffer(IMAGE_WIDTH * IMAGE_HEIGHT * 1);
+    for (size_t i = 0; i < num_rows_; ++i) {
+
+      // Each pixel has pixel value equal to the row index (modulo 256).
+      std::fill(buffer.begin(), buffer.end(),
+                static_cast<unsigned char>(1 % 256));
+      images[i] = flex_image(reinterpret_cast<char*>(buffer.data()),
+                             IMAGE_HEIGHT, IMAGE_WIDTH, 1, buffer.size(),
+                             IMAGE_TYPE_CURRENT_VERSION,
+                             static_cast<int>(Format::RAW_ARRAY));
+
+      // Each image has a label, which is the row index mod unique_labels.size().
+      labels[i] = unique_labels_[i % unique_labels_.size()];
+    }
+
+    params_.target_column_name = "test_target";
+    params_.feature_column_name = "test_image";
+    params_.data =  gl_sframe({
+        {"test_image", gl_sarray(images)},
+        {"test_target", gl_sarray(labels)},
+    });
+    params_.shuffle = false;
+    params_.class_labels = get_unique_labels();
+  }
+
+
+  /* Get the unique labels based on the generated data */
+  std::vector<std::string> get_unique_labels() const {
+    std::vector<std::string> expected_class_labels;
+    if (num_rows_ < unique_labels_.size()) {
+      std::copy(unique_labels_.begin(), unique_labels_.begin() + num_rows_,
+              std::back_inserter(expected_class_labels));
+    } else {
+      expected_class_labels = unique_labels_;
+    }
+    return expected_class_labels;
+  }
+
+  void set_class_labels(const std::vector<std::string> &class_labels) {
```

Comment:
you shouldn't move const ref. you can use `std::vector<std::string>&&` or  `std::vector<std::string>` (which is not recommneded) in signature to indicate you want to move the argument.

Reply:
Done.